### PR TITLE
Follow absolute path when redirect location contains ".." (#582)

### PR DESCRIPTION
**Pull request by jmahoney**
This is a fix for #582. First time contributor so keen for feedback.

### CODE REVIEW

File: lib/httparty/request.rb [hunk 1]
```diff
@@ -92,6 +92,11 @@ def uri
 
         path.path = "/#{path.path}" if last_uri_host[-1] != "/"
         path.path = last_uri_host + path.path
+
+        # use absolute path [#582]
+        if path.path.include?('/../')
```
Comment (TheSmartnik) [inline lib/httparty/request.rb]: Should we account for paths like `./foo/bar` as well?

File: lib/httparty/request.rb [hunk 2]
```diff
@@ -92,6 +92,11 @@ def uri
 
         path.path = "/#{path.path}" if last_uri_host[-1] != "/"
         path.path = last_uri_host + path.path
+
+        # use absolute path [#582]
```
Comment (TheSmartnik) [inline lib/httparty/request.rb]: I think this comment is unnecessary, people could you `git blame` to figure out why this change was added

File: spec/httparty/request_spec.rb [hunk 3]
```diff
@@ -664,6 +664,18 @@
           expect(response.parsed_response).to eq({"hash" => {"foo" => "bar"}})
         end
 
+        it "redirects to an absolute path if a 300 contains a relative path in the location header" do
+          redirect = stub_response '', 300
+          redirect['location'] = '../bar/baz'
+          ok = stub_response('<hash><bar>baz</bar></hash>', 200)
+          allow(@http).to receive(:request).and_return(redirect, ok)
+          response = @request.perform
```
Comment (TheSmartnik) [inline spec/httparty/request_spec.rb]: This needs to be surrounded by newlines from above and below to visually separate the test into 3 sections _Arrange, Act, Assert_

File: lib/httparty/request.rb (hunk 1 above)
Comment (jmahoney) [reply lib/httparty/request.rb]: It's a slightly different case but happy to tackle it if you think it's worth it?  

Given a request to https://test.com/v1
 
a redirect to a relative path of "../foo/bar" should result in a new request to https://test.com/foo/bar - the foo directory is a sibling of /v1

a redirect to a  relative path of "./foo/bar" should result in a new request to https://test.com/v1/foo/bar - the foo directory is a child of v1

File: spec/httparty/request_spec.rb (hunk 3 above)
Comment (jmahoney) [reply spec/httparty/request_spec.rb]: Done

File: lib/httparty/request.rb (hunk 2 above)
Comment (jmahoney) [reply lib/httparty/request.rb]: Removed

**Comment by jnunemaker** [comment]
@jmahoney I dropped the ball on this. Apologies. Happy to merge if you can get it up to date. Closing as stale in the meantime.
